yarn-project/simulator/src/public/public_execution_context.ts

```diff
@@ -56,14 +58,15 @@ export class PublicExecutionContext extends TypedOracle {
    */
   public getInitialWitness(witnessStartIndex = 0) {
     const { callContext, args } = this.execution;
-    const fields = [
-      ...callContext.toFields(),
-      ...this.header.toFields(),
-      ...this.globalVariables.toFields(),
-      new Fr(this.sideEffectCounter.current()),
-      ...args,
-    ];
-
+    const publicContextInputs = new PublicContextInputs(
+      callContext,
+      this.header,
+      this.globalVariables,
+      this.sideEffectCounter.current(),
+      Gas.test(), // TODO(palla/gas): Set proper values
+      new Fr(0),
```

Comment:
Just to make sure I understand correctly, this will be non-zero in teardown, right? Will it come as a parameter to the execution context?

Reply:
Yep, exactly!